topdown/query.go

```diff
@@ -225,6 +233,10 @@ func (q *Query) PartialRun(ctx context.Context) (partials []ast.Body, support []
 	}
 	f := &queryIDFactory{}
 	b := newBindings(0, q.instr)
+	builtinCache := builtins.Cache{}
+	if !q.time.IsZero() {
+		builtinCache.Put(nowKey, ast.NumberTerm(int64ToJSONNumber(q.time.UnixNano())))
```

Comment:
Out of curiosity, why in the `builtinCache` rather than as a new field on the `BuiltinContext`?

If we move it from being in the cache and set it at `eval` init time versus first time calling `time.now_ns` it might simplify some of the implementation for the builtin. It basically just needs to then blindly return `bctx.time` or whatever rather than mess with the caching stuff.

Reply:
No reason in particular. I've refactored it as you described. 👍